storage/client_merge_test.go

```diff
@@ -246,7 +248,9 @@ func TestStoreRangeMergeNonConsecutive(t *testing.T) {
 	// See #702 and TestStoreExecuteCmdOutOfRange.
 	// TODO(bdarnell): refactor this test to rebalance the range onto a separate node
 	// when this is supported.
-	rangeB.WaitForLeaderLease(t)
+	util.SucceedsWithin(t, 1*time.Second, func() error {
```

Comment:
can you move this retry loop back where it was and just use `util.IsTrueWithin`? That should eliminate the testing dependency.